src/backend/distributed/commands/alter_table.c

```diff
@@ -640,6 +640,8 @@ ConvertTable(TableConversionState *con)
 		Oid partitionRelationId = InvalidOid;
 		foreach_oid(partitionRelationId, partitionList)
 		{
+			char *tableQualifiedName = generate_qualified_relation_name(
+				partitionRelationId);
```

Comment:
I think we can drop this variable:

con->colocateWith = generate_qualified_relation_name(partitionRelationId);

Reply:
I think we can't. We need to get a qualified name before table conversion happened. If we move this line to L701, we can not reach the object by its relationId, because it's dropped.

Reply:
ah you're right